src/crs/basic.jl

```diff
@@ -142,12 +142,11 @@ function tol(coords::Cartesian)
   atol(numtype(Q)) * unit(Q)
 end
 
-Random.rand(rng::Random.AbstractRNG, ::Random.SamplerType{Cartesian{Datum,N}}) where {Datum,N} =
-  Cartesian{Datum}(ntuple(i -> rand(rng), N)...)
-
-Random.rand(rng::Random.AbstractRNG, ::Random.SamplerType{Cartesian2D}) = rand(rng, Cartesian2D{NoDatum})
-
-Random.rand(rng::Random.AbstractRNG, ::Random.SamplerType{Cartesian3D}) = rand(rng, Cartesian3D{NoDatum})
+makeconcrete(T::Type{Cartesian{Datum,N,L}}) where {Datum,N,L} = T
+makeconcrete(::Type{Cartesian{Datum,N}}) where {Datum,N} = 
+  Cartesian{Datum,N,Met{Float64}} |> makeconcrete
+makeconcrete(::Type{Cartesian3D}) = Cartesian{NoDatum, 3} |> makeconcrete
+makeconcrete(::Type{Cartesian2D}) = Cartesian{NoDatum, 2} |> makeconcrete
```

Comment:
Our design assumes `CRS{Datum}` and `CRS` as the only union-all types used by end-users. This means that we don't necessarily need to support `rand(ConcreteCRSType)`. Please remove the `makeconcrete` helper.

Reply:
I was still doing some additional benchmarking, but the use of a concrete constructor inside the `rand` method (as opposed to the generic non-concrete fallback) does significantly speed up things, like 6 times for the generation of LatLon coordinates for example.

That was the reason for having this option